src/main/kotlin/com/terraformation/backend/search/api/SearchController.kt

```
@@ -153,7 +153,7 @@ class SearchController(
   private fun resolvePrefix(prefix: String?): SearchFieldPrefix {
     val table =
         if (prefix != null) {
-          organizationsTable.resolveTable(prefix)
+          searchTables[prefix] ?: organizationsTable.resolveTable(prefix)
         } else {
```

Technically this affects some search tables whose name matches their sublist name in the `OrganizationsTable`, e.g. `batches`, `draftPlantingSites`, `facilities`, `inventories`, etc because of the join logic. Whether that's an issue or not for existing queries, I'm not sure 🤔

---

But my PR description could make that distinction clearer.

---

This is only true because the searchTable name for BatchesTable is `batches`, and the sublist for batches on the OrganizationsTable is also `batches`. I think technically to not break anything you'd have to flip the line to this:
```kotlin
organizationsTable.resolveTableOrNull(prefix) ?: searchTables[prefix]
```

Not necessarily suggesting that because it has other issues since you would still want to throw an error if neither of them found it, just mentioning it as an example.

---

`resolvePrefix()` doesn't return any context information about how the prefix resolved to a particular search table, though; it just wraps a single `SearchTable` in a `SearchPrefix` that has no sublists. A request of
```
"prefix": "batches",
"fields": ["batchNumber"]
```
with no search filters will return all visible batch numbers with or without this change (just tried it on staging to make sure I wasn't missing something).

---

Yup you're absolutely right, I totally forgot that it's just setting the root table for the prefix, not getting the table relative to the OrganizationsTable